Flakey test: Test_PushMetrics/WAL/staleNaNDoubleGauge_case

As seen here: https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/5484352499?check_suite_focus=true

```
=== CONT  Test_PushMetrics/WAL/staleNaNDoubleGauge_case
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd1fd6c]

goroutine 115 [running]:
github.com/tidwall/wal.(*Log).Read(0x0, 0x1)
	/home/runner/go/pkg/mod/github.com/tidwall/wal@v1.1.7/wal.go:621 +0x6c
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter.(*prweWAL).readPrompbFromWAL(0xc0001274a0, {0x1080f58, 0xc0003f7c00}, 0x0)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter/wal.go:348 +0x1ae
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter.(*prweWAL).continuallyPopWALThenExport(0xc0001274a0, {0x1080f58, 0xc0003f7c00}, 0xf7fc00)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter/wal.go:233 +0x3bf
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter.(*prweWAL).run.func1()
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter/wal.go:176 +0x185
created by github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter.(*prweWAL).run
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter/wal.go:166 +0x3e5
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter	45.234s
```

--- a/exporter/prometheusremotewriteexporter/wal.go
+++ b/exporter/prometheusremotewriteexporter/wal.go
@@ -165,12 +165,12 @@ func (prwe *prweWAL) run(ctx context.Context) (err error) {
 	waitUntilStartedCh := make(chan bool)
 	go func() {
 		signalStart := func() { close(waitUntilStartedCh) }
+		defer cancel()
 		for {
 			select {
 			case <-runCtx.Done():
 				return
 			case <-prwe.stopChan:
-				cancel()
 				return
 			default:
 				err := prwe.continuallyPopWALThenExport(runCtx, signalStart)
@@ -338,11 +338,17 @@ func (prwe *prweWAL) readPrompbFromWAL(ctx context.Context, index uint64) (wreq 
 		select {
 		case <-ctx.Done():
 			return nil, ctx.Err()
+		case <-prwe.stopChan:
+			return nil, fmt.Errorf("attempt to read from WAL after stopped")
 		default:
 		}
 
 		if index <= 0 {
 			index = 1
+		}
+
+		if prwe.wal == nil {
+			return nil, fmt.Errorf("attempt to read from closed WAL")
 		}
 
 		protoBlob, err = prwe.wal.Read(index)

